src/main/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlService.java

```diff
@@ -228,17 +240,27 @@ public CaseEventTrigger setReadOnlyOnCaseViewFieldsIfNoAccess(final CaseEventTri
                     CaseField field = caseFieldOpt.get();
                     if (!hasAccessControlList(userRoles, access, field.getAccessControlLists())) {
                         caseViewField.setDisplayContext(READONLY);
+                        setReadonlyOnWizardPageFieldDisplayContext(caseEventTrigger.getWizardPages(), caseViewField.getId());
                     }
                     if (field.isCompound()) {
                         setChildrenAsReadOnlyIfNoAccess(caseEventTrigger.getWizardPages(), field.getId(), field, access, userRoles, caseViewField);
                     }
                 } else {
                     caseViewField.setDisplayContext(READONLY);
+                    setReadonlyOnWizardPageFieldDisplayContext(caseEventTrigger.getWizardPages(), caseViewField.getId());
                 }
             });
         return caseEventTrigger;
     }
 
+    private void setReadonlyOnWizardPageFieldDisplayContext(List<WizardPage> wizardPages,
+                                                            String caseViewFieldId) {
+       wizardPages.forEach(wizardPage -> {
+           Optional<WizardPageField> wizardPageField = wizardPage.getWizardPageField(caseViewFieldId);
+           wizardPageField.ifPresent(field -> field.setDisplayContext(READONLY));
```

Comment:
This fix would work. But I think it's not the optimal fix. This is a regression on a feature that worked in the past. The bug is happening because recently we have done some changes (not sure if by mistake or voluntarily) to the display context of a field. 
In the original CCD design the display context of a field **for an event** is stored on the case field directly (the various`CaseField` classes in the toolkit and Data Store backend). 
When we remove update permission Data Store sets the display context of that event field as Readonly.
2 things changed recently:
1) a duplicate display context property has been added to the WizardPageField which is returned to the UI together inside an event field.
2) the UI has been changed to copy the WizardPageField display context to the event field, basically ignoring and overwriting the value Data Store is returning.
But Data Store has not been updated to also change this additional display context on update permission removal. Data store is thereby returning to the UI two display context values not in sync. 
This fix solves the problem by making sure the 2 display contexts have the same value.
But in my opinion the correct fix would be to remove this duplication by keeping only 
one of them. Keeping these 2 display contexts is increasing the complexity of Data Store and the UI for no reason. And this bug is a proof. Probably safer to keep the one on the case field to minimise the impact of the change. 
This will make sure we avoid similar bugs in Data Store in the future because there's only one place that controls the display context of a field.